packages/flutter_tools/lib/src/reporting/crash_reporting.dart

```diff
@@ -26,15 +26,70 @@ const String _kStackTraceFileField = 'DartError';
 /// it must be supplied in the request.
 const String _kStackTraceFilename = 'stacktrace_file';
 
+class CrashDetails {
+  CrashDetails({
+    @required this.command,
+    @required this.error,
+    @required this.stackTrace,
+    @required this.doctorText,
+  });
+
+  final String command;
+  final dynamic error;
+  final StackTrace stackTrace;
+  final String doctorText;
+}
+
+/// Reports information about the crash to the user.
+class CrashReporter {
+  CrashReporter({
+    @required this.fileSystem,
+    @required this.logger,
+    @required this.flutterProjectFactory,
+    @required this.client,
+  });
+
+  final FileSystem fileSystem;
```

Comment:
If there isn't some reason in this PR that these need to be public, then following suit with this pattern in the rest of the tool, they should be private.

Reply:
Okay, I'll make them private.